src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/submitter.py

```diff
@@ -65,40 +65,74 @@ def _get_scene_asset_references(rop_node: hou.Node) -> AssetReferences:
 
-    rop_dir_map = {
-        # Mantra/karma
-        "Driver/ifd": "vm_picture",
-        "Driver/karma": "picture",
-        # Husk
-        "Lop/usdrender_rop": _husk_outputs,
-        # Arnold
-        "Driver/arnold": "ar_picture",
-        # Vray
-        "Driver/vray_renderer": "SettingsOutput_img_file_path",
-        "Sop/rop_vrayproxy": "filepath",
-        "Driver/rop_vrayproxy": "filepath",
-        # Renderman
-        "Driver/ris::3.0": _renderman_outputs,
-        # Redshift
-        "Driver/Redshift_ROP": "RS_outputFileNamePrefix",
-        # Geo
-        "Driver/geometry": "sopoutput",
-        "Driver/alembic": "filename",
-        "Sop/filecache": "file",
-    }
-    # collect output dirs for each ROP
-    all_inputs = rop_node.inputAncestors()
-    for node in all_inputs:
+    asset_references.output_directories.update(_get_output_directories(rop_node))
+
+    return asset_references
+
+
+def _get_output_directories(rop_node: hou.Node) -> set[str]:
+    """
+    Find and return the set of all output directories detected from the output
+    paths within the nodes leading into the Deadline Cloud ROP node
+    """
+
+    def find_output_dirs(node: hou.Node) -> set[str]:
+        """
+        Given a node, find and return the directories of the output path parms.
+        """
+        out_dirs = set()
         type_name = node.type().nameWithCategory()
         out_parm = rop_dir_map.get(type_name, None)
-        if out_parm is not None:
+
+        if out_parm:
             if callable(out_parm):
-                computed_dirs = out_parm(node)
-                asset_references.output_directories.update(computed_dirs)
+                if type_name == "Driver/wedge":
+                    node = node.node(node.parm("driver").eval())
+                elif type_name == "Driver/fetch":
+                    node = node.node(node.parm("source").eval())
+
+                if node:
+                    detected_dirs = out_parm(node)
+                    out_dirs.update(detected_dirs)
             else:
                 path = node.parm(out_parm).eval()
-                asset_references.output_directories.add(os.path.dirname(path))
-    return asset_references
+                out_dirs.add(os.path.dirname(path))
+
+        return out_dirs
+
+    rop_dir_map = {
+        "Driver/alembic": "filename",  # Alembic
+        "Driver/arnold": "ar_picture",  # Arnold
+        "Driver/baketexture::3.0": "vm_uvoutputpicture1",  # Bake Texture
+        "Driver/channel": "chopoutput",  # Channel
+        "Driver/comp": "copoutput",  # Composite
+        "Driver/dop": "dopoutput",  # Dynamics
+        "Driver/fetch": find_output_dirs,  # Fetch
+        "Sop/filecache": "file",  # File Cache
+        "Sop/filecache::2.0": "file",  # File Cache v2
+        "Driver/filmboxfbx": "sopoutput",  # Filmbox FBX
+        "Driver/geometry": "sopoutput",  # Geometry
+        "Lop/usdrender_rop": _husk_outputs,  # Husk
+        "Driver/karma": "picture",  # Karma
```

Comment:
I believe Karma's node type is `Driver/usdrender`

Reply:
It may depend on if you're grabbing output vs. other settings? Some things will be on the Karma node, and others may be in the inner usdrender node. @lucaseck can confirm when they see this

Reply:
Yup, can confirm that just adding the Karma node it appears as `Driver/karma` when querying the node using `.type().nameWithCategory()`

Reply:
Good point though that USD Render is missing. I'll add that one in.

Reply:
To the original point though. The output information **_is_** located on `Driver/karma`?

Reply:
Yes, the output information is on `Driver/karma` when using the Karma node.

Reply:
Ended up leaving out adding in USD Render for now. It needs more digging into how output paths work within it.